### PR TITLE
[RFC] Update options to remove compatibility settings and have more sane defaults

### PR DESCRIPTION
**Pull request by c-f-h**
Main changes:
- the `vimoption_T` struct now only has a single `def_val` field instead of one for Vi defaults and one for Vim defaults. I used the Vim defaults wherever they diverged.
- I changed the defaults for most options to match tpope's [sensible.vim](https://github.com/tpope/vim-sensible)
- updated documentation to reflect the new defaults and remove any references to Vi defaults

I consider this PR finished and ready to merge, unless there are comments.

### CODE REVIEW

File: runtime/doc/options.txt [hunk 1]
```diff
@@ -45,9 +45,9 @@ achieve special effects.  These options come in three forms:
 :se[t] inv{option}	Toggle option: Invert value. {not in Vi}
 
 				*:set-default* *:set-&* *:set-&vi* *:set-&vim*
-:se[t] {option}&	Reset option to its default value. {not in Vi}
-:se[t] {option}&vi	Reset option to its Vi default value. {not in Vi}
-:se[t] {option}&vim	Reset option to its Vim default value. {not in Vi}
+:se[t] {option}&	Reset option to its default value.
+:se[t] {option}&vi	Reset option to its default value.
+:se[t] {option}&vim	Reset option to its default value.
```
Comment (ZyX-I) [inline runtime/doc/options.txt]: Do not repeat text three times. I would suggest to remove first description completely, remove second line at all (both description and `:set` command) and leave only the last line as it is now:

```
:se[t] {option}&
:se[t] {option}&vim    Reset option to its default value.
```

**Comment by justinmk** [comment]
Please see 

- https://github.com/neovim/neovim/pull/228 
- https://github.com/neovim/neovim/pull/228#issuecomment-67878227
- https://github.com/neovim/neovim/pull/1667

**Comment by c-f-h** [comment]
The cp option is already forced off, so we're effectively removing dead code with this. Dead code is always a maintenance problem. Besides, the code is several hundred lines shorter after this. I'd say this definitely reduces complexity.

What is the real world use case for set-&vi?

File: src/nvim/option.c [hunk 2]
```diff
@@ -2131,10 +2013,10 @@ void set_string_default(const char *name, const char_u *val)
   int opt_idx = findoption((char_u *)name);
   if (opt_idx >= 0) {
     if (options[opt_idx].flags & P_DEF_ALLOCED) {
-      free(options[opt_idx].def_val[VI_DEFAULT]);
+      free(options[opt_idx].def_val);
     }
 
-    options[opt_idx].def_val[VI_DEFAULT] = (char_u *) xstrdup((char *) val);
+    options[opt_idx].def_val = (char_u *) xstrdup((char *) val);
```
Comment (splinterofchaos) [inline src/nvim/option.c]: Are we moving away from `vim_strsave()`? If not, `vim_strsave(val)` wouldn't need any casts.

File: src/nvim/option.c [hunk 3]
```diff
@@ -2535,13 +2416,12 @@ do_set (
 
       if (vim_strchr((char_u *)"?=:!&<", nextchar) != NULL) {
         arg += len;
-        cp_val = p_cp;
+        /* These suffixes do nothing in Nvim, but we keep them for
+         * backwards compatibility for now. */
```
Comment (splinterofchaos) [inline src/nvim/option.c]: C++ style comments, please (`//`).

File: runtime/doc/options.txt [hunk 4]
```diff
@@ -4956,7 +4953,7 @@ A jump table for the options with a short description can be found at |Q_op|.
 	Negative or zero value means no thread scheduling.
 
 							*'nrformats'* *'nf'*
-'nrformats' 'nf'	string	(default "octal,hex")
+'nrformats' 'nf'	string	(default "hex")
```
Comment (splinterofchaos) [inline runtime/doc/options.txt]: Why remove "octal"?

File: runtime/doc/options.txt [hunk 5]
```diff
@@ -1624,7 +1624,7 @@ A jump table for the options with a short description can be found at |Q_op|.
 	"set nocp" is ignored, and "set cp" is an error.
 
 						*'complete'* *'cpt'* *E535*
-'complete' 'cpt'	string	(default: ".,w,b,u,t,i")
+'complete' 'cpt'	string	(default: ".,w,b,u,t")
```
Comment (splinterofchaos) [inline runtime/doc/options.txt]: Why remove scanning included files?

**Comment by justinmk** [comment]
The way forward regarding `'cpo'` and changing the defaults to sensible.vim values is to do so one at a time. This PR isn't going to be merged for that reason.

When removing a Vi-default, the corresponding `'cpo'` option should be removed, and any logic that exists to support it. This is where the real gains will be: removing logic that only exists to support `'cpo'` behaviors. For example, there have been many bugs related to `'linbreak'` and `'showbreak'` that exist only with certain combinations of `'cpo'` options.

Maybe some `'cpo'` behaviors are worth keeping, I don't know. It doesn't really matter right now because removing them all at once is not feasible. Each removal will need a separate PR/discussion.

**Pull request closed without merging**